src/maxdiffusion/train_utils.py

```diff
@@ -154,3 +159,33 @@ def generate_timestep_weights(config, num_timesteps):
   weights[bias_indices] *= timestep_bias_config["multiplier"]
   weights /= weights.sum()
   return jnp.array(weights)
+
+def save_checkpoint(save_fn, params, config, output_dir):
+  """
+  Save checkpoint.
+
+  Args:
+    save_fn: Must be a save_pretrained fn, for example, pipeline.save_pretrained.
+    params: params to save.
+    config: pyconfig
+    output_dir: local directory path.
+  """
+  user_dir = os.path.expanduser("~")
+  local_output_dir = output_dir.replace(os.path.join(
+    config.base_output_directory,
+    config.run_name
+  ), user_dir)
+
+  save_fn(
+    local_output_dir,
+    params=params
+  )
+
+  if jax.process_index() == 0 and config.upload_ckpts_to_gcs:
+    max_utils.walk_and_upload_blobs(config, local_output_dir)
+    # delete files in output_dir to save space
+    max_logging.log(f"Deleting {local_output_dir} to save space.")
+    shutil.rmtree(local_output_dir)
+
+  return local_output_dir
```

Comment:
this returned dir is not used, and will be invalid if we delete it in our code.

Reply:
That is correct, I left it as mlperf_4 branch, should I remove it?